### PR TITLE
Fix links in concepts section

### PR DESCRIPTION
**Pull request by tengqm**
We are still referencing the redirect entries in the docs, at many places.
These references should be updated to the correct content page.

### CODE REVIEW

File: content/en/docs/concepts/services-networking/service.md [hunk 1]
```diff
@@ -1047,7 +1045,7 @@ spec:
 
 Using the userspace proxy for VIPs, work at small to medium scale, but will
 not scale to very large clusters with thousands of Services.  The [original
-design proposal for portals](http://issue.k8s.io/1107) has more details on
+design proposal for portals](https://issue.k8s.io/1107) has more details on
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md:1048]: Is it better to link to https://github.com/kubernetes/kubernetes/issues/1107?

(If not, maybe we should change all the other issue links to use issue.k8s.io - could be a separate PR of course).

File: content/en/docs/concepts/storage/volumes.md [hunk 2]
```diff
@@ -18,10 +18,7 @@ Container starts with a clean state.  Second, when running Containers together
 in a `Pod` it is often necessary to share files between those Containers.  The
 Kubernetes `Volume` abstraction solves both of these problems.
 
-Familiarity with [Pods](/docs/user-guide/pods) is suggested.
-
-
-
+Familiarity with [Pods](/docs/concepts/workloads/pods/pod/) is suggested.
```
Comment (sftim) [inline content/en/docs/concepts/storage/volumes.md:21]: I would use a glossary tooltip here; however, this is fine.

File: content/en/docs/concepts/services-networking/service.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/concepts/services-networking/service.md:1048]: suggestion taken.

Comment (tengqm) [reply content/en/docs/concepts/services-networking/service.md:1048]: btw, this shortened url does look nice, when compared to `https://github.com/kubernetes/kubernetes/issues/1107`.

**Comment by jimangel** [comment]
Thanks @tengqm! Please rebase when time permits!
